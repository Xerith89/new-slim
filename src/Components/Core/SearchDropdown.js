import React, { Component } from 'react'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default class SearchDropdown extends Component {

    constructor(props){
        super(props)

        this.state = {
            type: '',
            number: ''
        }
    }

    onChange = (event) => {
        let name = event.target.name;
        let target = event.target.value;
        this.setState({
            [name]: target
        });
    }

    handleSearch = (event) => {
        event.preventDefault();
        console.log("Make request to API");
    }

    render() {
        return (
            <div data-testid="searchMenu">
                <div className="list-group dropdown-menu dropdown-menu-right mr-5 text-center">
                    <h5><FontAwesomeIcon icon={faSearch}/> Quick Search</h5>
                    <form className="m-2">
                        <div className="form-group" onChange={this.onChange}>
                            <input type="radio" value="spec" name="type" defaultChecked /> Specification 
                            <input type="radio" value="claim" name="type" className="ml-2"/> Claim
                        </div>
                        <div className="form-group">
                            <label>Search Number</label>
                            <input type="text" onChange={this.onChange} name="number" className="form-control"/>
                        </div>
                        <button className="btn btn-block btn-primary" onClick={this.handleSearch} >Search</button>
                    </form>
                    <button className="bg-transparent border-0 link">Advanced Search</button>
                </div>
            </div>
        )
    }
}
