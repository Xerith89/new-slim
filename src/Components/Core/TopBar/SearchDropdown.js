import React, { Component } from 'react'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default class SearchDropdown extends Component {

    constructor(props){
        super(props)

        this.state = {
            type: 'Spec',
            number: '',
        }

    }

    handleChange = (event) => {
        let name = event.target.name;
        let target = event.target.value;
        this.setState({
            [name]: target}
        );
    }

    handleSearch = (event) => {
        event.preventDefault();
        if (this.state.number !== "") {
            console.log("Make request to API");
        }
    }

    render() {
        return (
            <div data-testid="searchMenu">
                <div className="list-group dropdown-menu dropdown-menu-right mr-5 text-center">
                    <h5><FontAwesomeIcon icon={faSearch}/> Quick Search</h5>
                    <form className="m-2">
                        <div className="form-group" onChange={this.handleChange}>
                            <input type="radio" value="Spec" name="type" defaultChecked /> Specification 
                            <input type="radio" value="Claim" name="type" className="ml-2"/> Claim
                        </div>
                        <div className="form-group">
                            <label>Search Number</label>
                            <input type="text" data-testid="searchText" onChange={this.handleChange} value={this.state.number} name="number" className="form-control"/>
                        </div>
                        <button className="btn btn-block btn-primary" onClick={this.handleSearch} data-testid="searchButton">Search</button>
                    </form>
                    <div className="panel panel-default">
                    </div>
                    <button className="bg-transparent border-0 link">Advanced Search</button>
                </div>
            </div>
        )
    }
}
