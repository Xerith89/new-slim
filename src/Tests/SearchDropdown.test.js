import React from 'react';
import SearchDropdown from '../Components/Core/TopBar/SearchDropdown';
import {render, cleanup, fireEvent} from '@testing-library/react';

afterEach(cleanup);

test('Search dropdown component renders to the DOM', () => {
    const {getByTestId} = render(<SearchDropdown />);
    expect(getByTestId('searchMenu')).toMatchSnapshot();
});

test('Update function is called when text entered into searchbox', () => {
    const spy = jest.fn();
    const {getByTestId} = render(<SearchDropdown handleChange={spy}/>);
    expect(getByTestId('searchText').value).toBe("");
    fireEvent.change(getByTestId('searchText'), { target: { value: "23" } });
    expect(getByTestId('searchText').value).toBe("23");
    //expect(spy).toHaveBeenCalledTimes(1);
});

test('Clicking search brings up results modal', () => {
    const {getByTestId, queryByTestId} = render(<SearchDropdown />);
    expect(queryByTestId('searchResults')).toBeFalsy();
    fireEvent.click(getByTestId('searchButton'));
    expect(queryByTestId('searchResults')).toBeTruthy();
});