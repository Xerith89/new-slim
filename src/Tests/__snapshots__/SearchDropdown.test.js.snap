// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search dropdown component renders to the DOM 1`] = `
<div
  data-testid="searchMenu"
>
  <div
    class="list-group dropdown-menu dropdown-menu-right mr-5 text-center"
  >
    <h5>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
       Quick Search
    </h5>
    <form
      class="m-2"
    >
      <div
        class="form-group"
      >
        <input
          checked=""
          name="type"
          type="radio"
          value="Spec"
        />
         Specification
        <input
          class="ml-2"
          name="type"
          type="radio"
          value="Claim"
        />
         Claim
      </div>
      <div
        class="form-group"
      >
        <label>
          Search Number
        </label>
        <input
          class="form-control"
          data-testid="searchText"
          name="number"
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          class="btn-block btn btn-dark"
          data-testid="searchButton"
          type="button"
        >
          Search
        </button>
      </div>
    </form>
    <div
      class="panel panel-default"
    />
    <button
      class="bg-transparent border-0 link"
    >
      Advanced Search
    </button>
  </div>
</div>
`;
